### PR TITLE
Validation for all data types

### PR DESCRIPTION
**Pull request by SwathiSankararaman**

### CODE REVIEW

File: lib/ucfg.rb [hunk 1]
```diff
@@ -43,20 +43,27 @@ def self.validate_recursively(config, schema, config_path)
     # fails if property is provided as other type
     config.each do |key, value|
       if schema["properties"].key?(key) && schema["properties"][key].key?("type")
-        if schema["properties"][key]["type"] != value_type(value)
+        if schema["properties"][key]["type"].include?(value_type(value))
```
Comment (orhantoy) [inline lib/ucfg.rb]: Initially I was a bit surprised this worked but I can see why now: both strings and arrays have an `include?` method. I have modified one of the specs so this branch is currently failing - let me know if it is not obvious what you need to change.

File: lib/ucfg.rb [hunk 2]
```diff
@@ -43,20 +43,27 @@ def self.validate_recursively(config, schema, config_path)
...
+         valid = true
+        elsif value.nil?
+          valid = false
+          errors << "Property `#{(config_path + [key]).join('.')}` must be of type `#{(schema['properties'][key]['type']).first}` or `#{(schema['properties'][key]['type']).last}` (provided `#{value_type(value)}`)"
```
Comment (orhantoy) [inline lib/ucfg.rb]: In the case of value being null, we can't expect there to be 2 types. I have modified one of the specs and you would need to make changes accordingly on this line.

File: lib/ucfg.rb [hunk 3]
```diff
@@ -43,20 +43,27 @@ def self.validate_recursively(config, schema, config_path)
...
     end
 
-    schema["properties"].each do |key, value|
-      if config[key].is_a?(Hash)
-        new_config = config[key]
-        new_schema = value
-        recursion_result = validate_recursively(new_config, new_schema, config_path + [key])
-        errors.concat(recursion_result.errors)
-        valid = false unless recursion_result.valid?
+    if(!schema["properties"].nil?)
```
Comment (orhantoy) [inline lib/ucfg.rb]: The parenthesis are not needed in Ruby, and I think something like the following would be a bit cleaner:

```suggestion
    if schema.key?("properties")
```

**Review by orhantoy — CHANGES_REQUESTED** [review]
Great work, Swathi 💪 

I added a few more specs to highlight some of the remaining issues/cases we need to handle as well. Let me know if my feedback is unclear or if you just have questions.

File: lib/ucfg.rb (hunk 3 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb]: Hmm yes I complicated with this...Changed

File: lib/ucfg.rb (hunk 1 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb]: Done

File: lib/ucfg.rb (hunk 2 above)
Comment (SwathiSankararaman) [reply lib/ucfg.rb]: Done

**Review by orhantoy — APPROVED** [review]
Great work, go ahead and merge 🎉

**Pull request merged**